core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -684,6 +687,27 @@ Cuda::size_type *cuda_internal_scratch_unified(const Cuda &instance,
 
 namespace Kokkos {
 
+bool Cuda::is_running() const {
+  switch (m_space_instance->m_internal_status) {
+    case Impl::ExecutionSpaceStatus::complete: return false;
+    case Impl::ExecutionSpaceStatus::submitted:
+      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaEventRecord(m_space_instance->m_last_event,
+                                                 m_space_instance->m_stream));
+      m_space_instance->m_internal_status = Impl::ExecutionSpaceStatus::running;
+      [[fallthrough]];
+    case Impl::ExecutionSpaceStatus::running:
+      if (cudaEventQuery(m_space_instance->m_last_event) == cudaSuccess) {
```

Comment:
This can't use the usual macro, I guess, because we're expecting an in-progress return, but any values other than that or success are still hard errors that we should handle accordingly.